app/lib/task/backend.dart

```diff
@@ -349,81 +344,84 @@ class TaskBackend {
       seen.removeWhere((_, updated) => updated.isBefore(since));
 
       // Wait until aborted or 10 minutes before scanning again!
-      await abort.future.timeout(Duration(minutes: 10), onTimeout: () => null);
+      await abort.future.timeoutWithClock(
+        Duration(minutes: 10),
+        onTimeout: () => null,
+      );
     }
   }
 
   Future<void> trackPackage(
     String packageName, {
     bool updateDependents = false,
   }) async {
-    var lastVersionCreated = initialTimestamp;
-    String? latestVersion;
-    final changed = await withRetryTransaction(_db, (tx) async {
-      // Lookup Package and PackageVersion in the same transaction.
+    // Lookup Package and PackageVersion in the same transaction.
+    final (package, packageVersions) =
+        await withRetryTransaction(_db, (tx) async {
       final packageFuture = tx.packages.lookupOrNull(packageName);
       final packageVersionsFuture =
           tx.versions.listVersionsOfPackage(packageName);
-      final stateFuture = tx.tasks.lookupOrNull(packageName);
+
       // Ensure we await all futures!
-      await Future.wait([packageFuture, packageVersionsFuture, stateFuture]);
-      final state = await stateFuture;
+      await Future.wait([
+        packageFuture,
+        packageVersionsFuture,
+      ]);
       final package = await packageFuture;
       final packageVersions = await packageVersionsFuture;
+      return (package, packageVersions);
+    });
 
-      if (package == null) {
-        return false; // assume package was deleted!
-      }
-      latestVersion = package.latestVersion;
+    if (package == null) {
+      return; // assume package was deleted!
+    }
 
-      // Update the timestamp for when the last version was published.
-      // This is used if we need to update dependents.
-      lastVersionCreated = packageVersions.map((pv) => pv.created!).max;
+    if (package.isNotVisible) {
+      await db.tasks.delete(runtimeVersion, packageName).execute();
+      return; // TODO: Purge caches
```

Comment:
Why not call `_purgeCache` (and remove the TODO) or, if needed, specify which cache need to be purged.

Reply:
I think this was a change I missed during migration.